src/Map/ScaleCombo/ScaleCombo.jsx

```diff
@@ -0,0 +1,207 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Select } from 'antd';
+const Option = Select.Option;
+import { isNumber, isEmpty } from 'lodash';
+
+import Logger from '../../Util/Logger';
+import MapUtils from '../../Util/MapUtil';
+
+/**
+ * Class representating a scale combo to choose map scale via a dropdown menu.
+ *
+ * @class The FloatingMapLogo
+ * @extends React.Component
+ */
+class ScaleCombo extends React.Component {
+
+  static propTypes = {
+    /**
+     * The zoomLevel.
+     * @type {Number}
+     */
+    zoomLevel: PropTypes.number,
+
+    /**
+     * The onZoomLevelSelect function. This function is passed to Select component
+     * @type {Function}
+     */
+    onZoomLevelSelect: PropTypes.func.isRequired,
+
+    /**
+     * The resolutions.
+     * @type {Array}
+     */
+    resolutions: PropTypes.arrayOf(PropTypes.number),
+
+    /**
+     * The scales.
+     * @type {Array}
+     */
+    scales: PropTypes.arrayOf(PropTypes.shape),
+
+    /**
+     * The style object
+     * @type {Object}
+     */
+    style: PropTypes.object,
+
+    /**
+     * The map
+     * @type {Ol.Map}
+     */
+    map: PropTypes.object
+  }
+
+  /**
+   * The default props
+   */
+  static defaultProps = {
+    style: {
+      width: 100
+    },
+    scales: []
+  }
+
+  /**
+   * Create a map.
+   * @constructs Map
+   */
+  constructor(props) {
+    super(props);
+
+    this.pushScaleOption = this.pushScaleOption.bind(this);
+    this.getOptionsFromMap = this.getOptionsFromMap.bind(this);
+    this.getVal = this.getVal.bind(this);
+    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
+    this.getInputElement = this.getInputElement.bind(this);
+    this.componentWillMount = this.componentWillMount.bind(this);
+
+    this.state = {
+      zoomLevel: props.zoomLevel
+    };
+  }
+
+  /**
+   * Unknown - Description
+   *
+   * @param {type} resolution Description
+   * @param {type} mv         Description
+   *
+   * @return {type} Description
+   */
+  pushScaleOption = (resolution, mv) => {
+    let scale = MapUtils.getScaleForResolution(resolution, mv.getProjection().getUnits());
+    let roundScale = Math.round(scale);
+    let option = <Option key={roundScale.toString()} value={roundScale.toString()}>1:{roundScale.toLocaleString()}</Option>;
+    this.props.scales.push(option);
```

Comment:
Props are immutable inside the component. Please make use of the state instead.

Reply:
Fixed in bef0bbd